### PR TITLE
Adds cf-atomic-component

### PR DESCRIPTION
**Pull request by anselmbradford**
Fixes https://github.com/cfpb/capital-framework/issues/585

## Additions

- Moves atomic-component micro-framework from its own repo to make it a module in CF as cf-atomic-component.

## Testing

- I'm not sure how to properly test this before it's published because `npm run cf-link` looks for the published `cf-atomic-component` on npm for `cf-expandables` and `cf-tables`. @contolini any ideas?

### CODE REVIEW

File: src/cf-atomic-component/package.json [hunk 1]
```diff
@@ -0,0 +1,14 @@
+{
+  "name": "cf-atomic-component",
+  "version": "1.4.0",
```
Comment (jimmynotjim) [inline src/cf-atomic-component/package.json]: Should this start below v1 or are we going to pick up where we left off when it was in it's own repo?

File: src/cf-atomic-component/src/mixins/Events.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    if ( events.hasOwnProperty( eventName ) === false ) {
+      return this;
+    }
+    for ( var i = 0, len = events[eventName].length; i < len; i++ ) {
+      this.events[eventName][i].apply( this, arguments );
+    }
+
+    return this;
+  }
+
```
Comment (jimmynotjim) [inline src/cf-atomic-component/src/mixins/Events.js]: Trivial: Remove extra new line

File: src/cf-atomic-component/src/utilities/config.js [hunk 3]
```diff
@@ -0,0 +1,51 @@
+/* ==========================================================================
+   Atomic configurations and constants
+
+   ========================================================================== */
+
+'use strict';
+
+// Bit values intended to be used for bit inversion.
+const DIRECTIONS = {
+  UP: 0,
```
Comment (jimmynotjim) [inline src/cf-atomic-component/src/utilities/config.js]: Should all these values be left aligned to match formatting in other files?

File: src/cf-atomic-component/src/utilities/data-set/index.js [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+'use strict';
+
+const _assign = require( '../object-assign' ).assign;
+
+/**
+ * Converts a string from selector-case to camelCase.
+ * @param {string} str - The string in selector-case form.
+ * @returns {string} The string in camelCase form.
+ */
+function _toCamelCase( str ) {
```
Comment (jimmynotjim) [inline src/cf-atomic-component/src/utilities/data-set/index.js]: Should we move this into it's own utility?

File: src/cf-atomic-component/package.json (hunk 1 above)
Comment (anselmbradford) [reply src/cf-atomic-component/package.json]: I think picking up where we left off would be clearest, but happy to set it any way that makes sense.

File: src/cf-atomic-component/src/mixins/Events.js (hunk 2 above)
Comment (anselmbradford) [reply src/cf-atomic-component/src/mixins/Events.js]: Fixed

File: src/cf-atomic-component/src/utilities/config.js (hunk 3 above)
Comment (anselmbradford) [reply src/cf-atomic-component/src/utilities/config.js]: Fixed

File: src/cf-atomic-component/src/utilities/data-set/index.js (hunk 4 above)
Comment (anselmbradford) [reply src/cf-atomic-component/src/utilities/data-set/index.js]: Prolly best to leave structural changes as a patch after it's released so it's easier to track the changes from the atomic-component repo to here.

File: src/cf-atomic-component/src/utilities/on-ready/index.js [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+
+/**
+ * Checks if the document is ready, if it is, trigger the passed function,
+ * if not, push the function to an array to be triggered after the page
+ * is loaded.
+ * @param {Function} fn -
+ *   Function to run only after the DOM has completely loaded.
+ * @returns {Array} List of functions to run after the DOM has loaded.
+ */
+function onReady( fn ) {
```
Comment (jimmynotjim) [inline src/cf-atomic-component/src/utilities/on-ready/index.js]: Outside scope, but could we change this to accept any elem, so it's useful whether we're waiting on the doc or an individual piece of the page?

File: src/cf-atomic-component/src/utilities/data-set/index.js (hunk 4 above)
Comment (jimmynotjim) [reply src/cf-atomic-component/src/utilities/data-set/index.js]: yep yep, makes sense.

File: src/cf-atomic-component/src/utilities/on-ready/index.js (hunk 5 above)
Comment (anselmbradford) [reply src/cf-atomic-component/src/utilities/on-ready/index.js]: Hmm, so it takes a dom node and waits till just that dom node is ready? I like it!

Comment (jimmynotjim) [reply src/cf-atomic-component/src/utilities/on-ready/index.js]: Yeah, it's the way jQuery ready works

**Pull request merged**